upload-api/config.js

```diff
@@ -9,11 +9,29 @@ import * as DID from '@ipld/dag-ucan/did'
  * @param {string} [config.UPLOAD_API_DID] - public DID for the upload service (did:key:... derived from PRIVATE_KEY if not set)
  * @returns {import('@ucanto/principal/ed25519').Signer.Signer}
  */
- export function getServiceSigner(config) {
+export function getServiceSigner(config) {
   const signer = ed25519.parse(config.PRIVATE_KEY)
   if (config.UPLOAD_API_DID) {
     const did = DID.parse(config.UPLOAD_API_DID).did()
     return signer.withDID(did)
   }
   return signer
 }
+
+/**
+ * Given a string, parse into provider ServiceDIDs. 
+ * 
+ * @param {string} providersEnvVar 
+ * @returns {import('@web3-storage/upload-api').ServiceDID[]}
+ */
+export function parseProviders(providersEnvVar) {
```

Comment:
This doesn't really have anything to do with providers - it's parsing a CSV of did:web prefixed DIDs. I'd call it something to that effect.

Reply:
Cool - renamed to parseServiceDids since that's what feels natural for the types. I think this could be further generalized to `parseDidWebs` or something but I do want to return `ServiceDIDs` in this case and have a ServiceDID-specific error message and don't feel like diving down the generics rabbithole right this sec, so feel pretty good about this level of abstraction for now.